microprofile/config-smallrye/src/main/java/org/wildfly/extension/microprofile/config/smallrye/ConfigSourceRootDefinition.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.microprofile.config.smallrye;
+
+import static org.jboss.as.controller.SimpleAttributeDefinitionBuilder.create;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.FILESYSTEM_PATH;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PersistentResourceDefinition;
+import org.jboss.as.controller.ReloadRequiredRemoveStepHandler;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.SimpleResourceDefinition;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+
+/**
+ * Resource definition to add a base directory as a config source. The contained top-level directories become
+ * dir type config sources.
+ *
+ * @author Kabir Khan
+ */
+class ConfigSourceRootDefinition extends PersistentResourceDefinition {
+
+    static AttributeDefinition ORDINAL = SimpleAttributeDefinitionBuilder.create("ordinal", ModelType.INT)
+            .setDefaultValue(new ModelNode(100))
```

Comment:
You are using the existing default ordinal version so no issue here.

However I wonder if we should maybe change it to avoid any clash with the microprofile-config.properties (that is also set to 100) and our existing config source definition.

Maybe putting it below 100 would make sure that a ConfigSourceDir would be use before such a ConfigSourceRoot? wdyt?

Reply:
@jmesnil I am refactoring as suggested to use a `root` nested attribute in the `dir` attribute of the original ConfigSourceDefinition. 

But your point got me thinking, and IMHO this ordinal should be **higher** than the microprofile-config.properties one so it fits in with the pattern of something outside the deployment being able to **override** the deployment values. 

I suggest 150, so that system properties and env vars can still override values set in the subsystem. Do you agree?